src/components/Form/Calculator/index.js

```diff
@@ -1,8 +1,75 @@
-const Calculator = () => {
+import {useEffect, useState} from "react";
+import Display from "./Display";
+import Results from "./Results";
+
+const Calculator = ({borrowAmount, annualSalary, monthlyRepay, click}) => {
+
+    const [toPayPerMonth, setToPayPerMonth] = useState(0)
+    const [payOffTimeMonths, setPayOffTimeMonths] = useState(0)
+    const [payOffYearly, setYearlyPayOff] = useState(0)
+    const [payOffTimeWholeYears, setPayOffTimeWholeYears] = useState(0)
+    const [payOffTimeExtraMonths, setPayOffTimeExtraMonths] = useState(0)
+
+    useEffect (() => {
+        let monthlySalary = annualSalary/12
+
+        let monthlyPercentageOfSalary = ((monthlySalary/100) * monthlyRepay)
+        let monthlyPercentageOfSalaryWithPence = monthlyPercentageOfSalary.toFixed(2)
+        setToPayPerMonth(parseInt(monthlyPercentageOfSalaryWithPence))
+
+        let monthlyPayOff = (borrowAmount/monthlyPercentageOfSalary).toFixed(0)
+        setPayOffTimeMonths(parseInt(monthlyPayOff))
+
+        let yearlyPayOff = monthlyPayOff/12
+        setYearlyPayOff(yearlyPayOff)
+        console.log(yearlyPayOff)
+
+        let wholeYears = Math.floor(yearlyPayOff)
+        setPayOffTimeWholeYears(parseInt(wholeYears))
```

Comment:
I believe you could remove the parseInt here (could be wrong though, I haven't tried!) because `wholeYears` is set by `Math.floor()` it should definitely be a number